packages/jazz-tools/src/tools/coValues/coList.ts

```diff
@@ -499,6 +302,301 @@ export class CoList<out Item = any> extends Array<Item> implements CoValue {
     return subscribeToCoValueWithoutMe<L, R>(this, id, options, listener);
   }
 
+  // Override mutation methods defined on Array, as CoLists aren't meant to be mutated directly
+  override push(...items: Item[]): number {
+    throw new Error(
+      "Cannot mutate a CoList directly. Use .$jazz.push(...) instead.",
+    );
```

Comment:
Overriding mutable Array methods to return an error will make the migration to `$jazz` more difficult to adopters, as they won't get type errors for unsupported methods.

I don't see a way to inherit only read-only array methods, though, so if we wanted CoList to only implement the `ReadonlyArray` interface, we'd need to reimplement all methods manually.

@gdorsi what do you think about this?

Reply:
> We can go with a warning, and leave things simple.

What do you mean by a warning? Log something to the console but still call the real method?

> As for protect us from undesired mutations, maybe we should get rid of the set part of the proxy?
This way the only way to really change the CoValue will be .$jazz
How much complexity would this add?

I think this is a similar case to the array mutation methods: throwing an error when calling set on the proxy is trivial, but will make the migration more difficult. I'd love direct assignments to throw a type error, but not sure how to do that without making big refactors on CoMap and CoList.

Reply:
Seems to do the job
<img width="1260" height="542" alt="Screenshot 2025-08-13 at 22 55 12" src="https://github.com/user-attachments/assets/6503c127-8aaf-4e87-bf1c-4bff3f2ce5b1" />